File: colorsample/src/main/java/com/tobiasschuerg/colorsample/MainActivity.java
package com.tobiasschuerg.colorsample;

import android.os.Bundle;
import android.support.annotation.ColorInt;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import com.tobiasschuerg.color.material.MaterialColor;
import com.tobiasschuerg.color.models.ColorModel;
import com.tobiasschuerg.color.models.HEXColor;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        MaterialColor materialColor = MaterialColor.random(true);
        toolbar.setBackgroundColor(materialColor.get500());
        toolbar.setTitleTextColor(materialColor.getTextBlackWhite());

        setStatusBarColor(materialColor.get700());


        TextView tv = (TextView) findViewById(R.id.hello);
        ColorModel color = new HEXColor("#ffff0000").toHSV().setSaturation(0.6f).setHue(190f).toHSL().setLightness(0.5f);
        tv.setTextColor(color.toColor());
    }

    private void setStatusBarColor(@ColorInt int color) {
        Window window = getWindow();
        // clear FLAG_TRANSLUCENT_STATUS flag:
        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
        // add FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS flag to the window
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        // finally change the color
        window.setStatusBarColor(color);
    }

}

File: colorsample/src/main/java/com/tobiasschuerg/colorsample/MainActivity.kt
package com.tobiasschuerg.colorsample

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.annotation.ColorInt
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton
import androidx.appcompat.widget.Toolbar
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePadding
import com.google.android.material.textfield.TextInputLayout
import com.tobiasschuerg.color.ColorCreator
import com.tobiasschuerg.color.helper.setAccentColor
import com.tobiasschuerg.color.material.MaterialColor
import com.tobiasschuerg.color.models.ColorModel
import com.tobiasschuerg.color.models.HEXColor
import com.tobiasschuerg.color.models.getTextBlackWhite

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rootView = findViewById<android.view.ViewGroup>(R.id.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        val contentLayout = findViewById<android.view.ViewGroup>(R.id.content_layout)

        // Handle window insets for edge-to-edge
        ViewCompat.setOnApplyWindowInsetsListener(rootView) { view, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())

            // Apply top inset to toolbar
            toolbar.updatePadding(top = systemBars.top)

            // Apply bottom inset to content
            contentLayout.updatePadding(bottom = systemBars.bottom)

            insets
        }

        setSupportActionBar(toolbar)

        val materialColor = MaterialColor(ColorCreator.randomColor())
        val color1 = materialColor.get500()
        toolbar.setBackgroundColor(color1.toColor())
        toolbar.setTitleTextColor(materialColor.getTextBlackWhite())

        // Set status bar color to transparent for edge-to-edge
        window.statusBarColor = android.graphics.Color.TRANSPARENT
        window.navigationBarColor = android.graphics.Color.TRANSPARENT

        val color: ColorModel = HEXColor("#ffaabbff")
                .toHSV()
                .copy(saturation = 0.6f)
                .copy(hue = 190f)
                .toHSL()
                .copy(lightness = 0.5f)

        val hello = findViewById<TextView>(R.id.hello)
        hello.setTextColor(color.toColor())

        val firstNameLayout = findViewById<TextInputLayout>(R.id.first_name_layout)
        val lastNameLayout = findViewById<TextInputLayout>(R.id.last_name_layout)
        firstNameLayout.setAccentColor(color1.toColor())
        lastNameLayout.setAccentColor(color1.toColor())

        val restartButton = findViewById<AppCompatButton>(R.id.restart_button)
        restartButton.setOnClickListener {
            startActivity(Intent(applicationContext, MainActivity::class.java))
            finish()
        }
    }
}
